### PR TITLE
Support policy tab

### PR DESCRIPTION
**Pull request by lahabana**
### Description

The Mesh team is generating their versions.yml from metadata and syncing it in this repo (see https://github.com/kumahq/kuma/pull/6421).
This adds the following features to leverage this:

- The policy table is dynamically generated from the versions.yml
- Add the possibility to have a recent version that isn't released yet by having the latest pointer be explicit
- We can explicitly ignore some release from rendering (useful for dev for example)

We also remove some old versions of the docs (1.0.x and 1.1x) which have been unsupported for a while and don't support this new automation.

### Testing instructions

Netlify link: https://deploy-preview-5374--kongdocs.netlify.app/mesh/latest/support-policy/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Review by fabianrbz — APPROVED** [review]
I left a small comment but otherwise, it looks good to me.
@lahabana can you check with one of the technical writers before merging just to confirm that it's ok to remove the old versions? we might have a process for deprecating docs that I'm not aware of.

File: app/_src/mesh/support-policy.md [hunk 1]
```diff
@@ -6,18 +6,9 @@ content-type: reference
...
-| Version  | Released Date | End of Full Support |
-|:--------:|:-------------:|:-------------------:|
-|  2.1.x   |  2023-02-01   |     2024-02-01      |
-|  2.0.x   |  2022-11-04   |     2024-11-03      |
-|  1.9.x   |  2022-08-24   |     2023-08-24      |
-|  1.8.x   |  2022-06-15   |     2023-06-14      |
-|  1.7.x   |  2022-04-11   |     2023-04-10      |
-|  1.6.x   |  2022-02-24   |     2023-02-23      |
-|  1.5.x   |  2021-11-23   |     2022-11-22      |
+{% support_policy kong_versions_mesh%}
```
[the remark this answers is not in the document]
Comment (lahabana) [reply app/_src/mesh/support-policy.md:11]: Actually this is meant to be the actual string "kong_versions_mesh" no?

[the remark this answers is not in the document]
Comment (fabianrbz) [reply app/_src/mesh/support-policy.md:11]: 🤦 you are right

**Comment by lena-larionova** [comment]
> As far as I know, we don't have a process for deprecating docs other than that we remove them from the nav. 

For Gateway versions before 2.0, we maintain an archive: https://legacy-gateway--kongdocs.netlify.app/enterprise/1.5.x/
However, this was mainly because we have a lot of Gateway customers still on very old versions. Unfortunately, we don't have analytics insight into how many people use this site, but I can guess that's it's probably very, very little, since there's already a huge dropoff between 3.x and 2.x, and even more so for anything before 2.8.

@lahabana if you have any concerns about people still using the older versions of Mesh, I'd create an archive. But honestly? I don't think the docs changed at all in the versions you deleted, and they certainly weren't very useful.

**Comment by lahabana** [comment]
Agree 100% with you

**Comment by lahabana** [comment]
Can I get this merged then? :)

**Pull request merged**